### PR TITLE
fix batch norm

### PR DESCRIPTION
**Pull request by piiswrong**
@winstywang @antinucleon @tornadomeet 
Current master doesn't set gamma to 1 during testing, in order to accommodate old models introduced before fix_gamma was introduced and set to true by default.

However this creates another problem as kvstore_local keeps a internal copy of all the weights so resetting to 1 inside bn is ignored and overwritten by kv's internal copy. As a result weight decay decreases gamma to 0 during training.

This PR fix this by resetting gamma whenever BN::Forward/Backward is called. However this invalidates all models trained before Dec 5 2015.

My suggestion is to merge this when we release 1.0 and explain this to the users in transition guide as well as provide a converter that set fix_gamma to false in .json symbols.

### CODE REVIEW

**Comment by tornadomeet** [comment]
agree!
i think we should reserve fix_gamma(fix_gamma=True is very useful instead of data mean-val normalization ), and set fix_gamma=False by default.

**Comment by pluskid** [comment]
Can you add an entry to [NEWS](https://github.com/dmlc/mxnet/blob/master/NEWS.md) under the ongoing v1.0 milestone and list it as breaking changes?
